wrapper.sh

```diff
@@ -1,120 +1,114 @@
-	grep '"Success":true' /tmp/MAM.output > /dev/null 2>/dev/null
-  	if [ $? -ne 0 ]
-  	then
-		echo response: `cat /tmp/MAM.output`
-		echo Current cookie file is invalid.  Please delete it, set the mam_id, and restart the container.
+	curl -s -b $COOKIE_FILE -c $COOKIE_FILE https://t.myanonamouse.net/json/dynamicSeedbox.php > $RESPONSE_FILE
+	grep '"Success":true' $RESPONSE_FILE > /dev/null 2>/dev/null
+  	if [ $? -ne 0 ]; then
+		echo "Response: `cat $RESPONSE_FILE`"
+		echo "Current cookie file is invalid.  Please delete it, set the mam_id, and restart the container."
 		exit 1
 	else
-		echo Session is valid
+		echo "Session is valid"
 	fi
 
 fi
 
-OLDIP=`cat $CACHEFILE 2>/dev/null`
+OLD_IP=`cat $OLD_IP_FILE 2>/dev/null`
 
-while [ $PPID -ne 1 ]
-do
-	OLDIP=`cat $CACHEFILE 2>/dev/null`
+while [ $PPID -ne 1 ]; do
+	OLD_IP=`cat $OLD_IP_FILE 2>/dev/null`
+	NEW_IP=`curl -s ip4.me/api/ | md5sum - | awk '{print $1}'`
 
-	NEWIP=`curl -s ip4.me/api/ | md5sum - | awk '{print $1}'`
-	if [ "x$DEBUG" != "x" ]
-	then
-		echo Current IP:  `curl -s ip4.me/api/`
+	if [ "x$DEBUG" != "x" ]; then
+		echo "Current IP:  `curl -s ip4.me/api/`"
 	fi
 	
 	# Check to see if the IP address has changed
-	if [ "${OLDIP}" != "${NEWIP}" ]
-	then
-  		echo New IP detected
-  		curl -s -b $COOKIEFILE -c $COOKIEFILE https://t.myanonamouse.net/json/dynamicSeedbox.php > /tmp/MAM.output
+	if [ "$OLD_IP" != "$NEW_IP" ]; then
+    echo "New IP detected"
+    # Save cookie jar to temporary file, to not overwrite
+    # with empty cookies on failed requests
+    curl -s -b $COOKIE_FILE -c $TEMP_COOKIE_FILE https://t.myanonamouse.net/json/dynamicSeedbox.php > $RESPONSE_FILE
 
-		grep -E 'No Session Cookie|Invalid session' /tmp/MAM.output > /dev/null 2>/dev/null
-		if [ $? -eq 0 ]
-		then
-			echo response: `cat /tmp/MAM.output`
-			echo Current cookie file is invalid.  Please delete it, set the mam_id, and restart the container.
+		grep -E 'No Session Cookie|Invalid session' $RESPONSE_FILE > /dev/null 2>/dev/null
+		if [ $? -eq 0 ]; then
+			echo "Response: `cat $RESPONSE_FILE`"
+			echo "Current cookie file is invalid.  Please delete it, set the mam_id, and restart the container."
 			exit 1
 		fi
 	
-  		# If that command worked, and we therefore got the success message
-  		# from MAM, update the CACHEFILE for the next execution
-		grep '"Success":true' /tmp/MAM.output > /dev/null 2>/dev/null
-  		if [ $? -eq 0 ]
-  		then
-			echo Response:  \"`cat /tmp/MAM.output`\"
-    			echo $NEWIP > $CACHEFILE
-			OLDPID=$NEWIP
+    # If that command worked, and we therefore got the success message
+    # from MAM, update the OLD_IP_FILE for the next execution
+		grep '"Success":true' $RESPONSE_FILE > /dev/null 2>/dev/null
+    if [ $? -eq 0 ]; then
+      echo "Response:  \"`cat $RESPONSE_FILE`\""
+      # Update COOKIE_FILE only on successful requests
+      mv $TEMP_COOKIE_FILE $COOKIE_FILE
```

Comment:
Other relevant part of this fix. Everything else is formatting or variable name changes.